### PR TITLE
Add GA Measurement Protocol client

### PR DESCRIPTION
**Pull request by njam**
Docu: https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters

Based on: https://github.com/cargomedia/CM/pull/1613
See also: https://github.com/cargomedia/fuboo/issues/2649

### CODE REVIEW

**Comment by njam** [comment]
When sending 1'000 `pageview` measurements in a loop each request takes 27ms!

screen shot 2015-01-29 at 10 07 54

cc @fauvel @moby86

**Comment by njam** [comment]
@fauvel please review

We can then use it like this:
```php
$googleAnalytics = new CMService_GoogleAnalytics_Client('UA-59766642-1');
$measurementProtocol = $googleAnalytics->getMeasurementProtocolClient();

$measurementProtocol->queueHit('event', [
    'uid' => '12345',
    'ec'  => 'Subscription',
    'ea'  => 'Create',
    'el'  => 'LTV',
    'ev'  => 12,
]);
```

**Comment by njam** [comment]
Changed it a bit. New usage:
```php
$googleAnalytics = new CMService_GoogleAnalytics_Client('UA-59766642-1');
$measurementProtocol = $googleAnalytics->getMeasurementProtocolClient();

$measurementProtocol->trackEvent([
    'userId'        => '12345',
    'eventCategory' => 'Subscription',
    'eventAction'   => 'Create',
    'eventLabel'    => 'LTV',
    'eventValue'    => 12,
]);
```

**Comment by njam** [comment]
@moby86 @fauvel 
According to [this blog post](http://www.cargomedia.ch/2015/02/18/googleanalytics-user-tracking.html) we can use *Measurement Protocol* to send user events independent from their browsers. Unfortunately they couldn't tell whether it will be useable in *AdWords Optimizer*.
I'd suggest to give it a try, wdyt?

**Comment by fauvel** [comment]
go go go!

**Comment by njam** [comment]
@fauvel did you check this PR? Is it good to merge?

File: library/CMService/GoogleAnalytics/MeasurementProtocol/SendHitJob.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+class CMService_GoogleAnalytics_MeasurementProtocol_SendHitJob extends CM_Jobdistribution_Job_Abstract {
+
+    protected function _execute(CM_Params $params) {
+        $propertyId = $params->getString('propertyId');
+        $parameterList = $params->getArray('parameterList');
+
+
```
Comment (fauvel) [inline library/CMService/GoogleAnalytics/MeasurementProtocol/SendHitJob.php]: Remove empty line

**Comment by fauvel** [comment]
As discussed, I think introducing an `aliasList` is a bit overkilled (I don't think Google will introduce several commands in the JavaScript API for the same measurement protocol parameter).

**Comment by njam** [comment]
I agree `aliasList` is probably not needed, but it doesn't add much complexity - only 1 or 2 lines of code. I'd keep it for the sake of completeness.
wdyt?

**Comment by fauvel** [comment]
agree! good to merge then.

**Pull request merged**